src/shared/components/ui/Form/DropdownSelect/DropdownSelect.js

```diff
@@ -0,0 +1,309 @@
+import Label from '@components/ui/Label'
+import Icon from '@components/ui/Icon'
+import { useStyles } from '@helpers/hooks/useStyles'
+import styles from './DropdownSelect.module.scss'
+
+/**
+ * Select - stateless presentational component
+ * @param {object} props - props
+ * @param {string} props.name - field name
+ * @param {string} props.value - field value
+ * @param {func} props.onChange - on Change handler
+ * @param {string} props.label - label
+ * @param {object} props.errors - element errors list
+ * @param {object} props.touched - element touched list
+ * @param {string} props.hasLabel - has label
+ * @param {array} props.options - list of options
+ * @param {func} props.handleOnSearchChange - on search input change handler
+ * @param {string} props.searchPlaceholder - search placeholder
+ * @param {bool} props.inModal - position in modal
+ * @param {bool} props.isLoading - list loading spinner
+ * @param {bool} props.isEmptyList - display empty list
+ * @param {bool} props.hasDescription - description
+ * @param {object} props.overflowStyle - overflow style
+ * @param {array} props.emphasisedOptions - emphasised options
+ * @param {bool} props.liveChanges - search by entering characters
+ * @param {bool} props.optionalContent
+ * @param {bool} props.hasLoadMoreButton - has load more button
+ * @param {function} props.loadMoreEvent - load more button event
+ * @param {object} props.i18n - object of translations
+ * @return {object} An object of children element
+ */
+const DropdownSelectForm = ({
+  name,
+  value,
+  onChange,
+  errors,
+  touched,
+  label,
+  hasLabel,
+  options,
+  handleOnSearchChange,
+  searchPlaceholder,
+  inModal,
+  isLoading,
+  isEmptyList,
+  hasDescription,
+  overflowStyle,
+  emphasisedOptions,
+  liveChanges,
+  optionalContent,
+  hasLoadMoreButton,
+  loadMoreEvent,
+  i18n
+}) => {
+  const [inModalPosition, setInModalPosition] = useState(false)
+
+  const labelStyles = useStyles({
+    [styles['form-field__label']]: true,
+    [styles['form-field__label--has-error']]: errors[name] || touched[name],
+    [styles['form-field__label--has-value']]: value[name]
+  })
+
+  const dropdownStyles = useStyles({
+    [styles['form-field__dropdown--has-error']]: errors[name] || touched[name]
+  })
+
+  const handleChange = useCallback((name, item) => {
+    emitCloseDropdown()
+    onChange(name, item)
+  })
+
+  useEffect(() => {
+    setInModalPosition(inModal)
+  }, [inModal])
+
+  const renderOption = item =>
+    hasDescription ? (
+      <Fragment>
+        <Button
+          variant='dropdown-element'
+          onClick={() => handleChange(name, item)}
+        >
+          <Heading level={5}>{item?.label}</Heading>
+
+          <Paragraph color='accent-2' size={12}>
+            {item?.description}
+          </Paragraph>
+        </Button>
+
+        <Divider />
+      </Fragment>
+    ) : (
+      <Button
+        variant='dropdown-element'
+        onClick={() => handleChange(name, item)}
```

Comment:
żaden linter się nie pluje o to?

Reply:
tak to było, więc zostawiam, żeby nie wybuchło, a linter jest do du.. skonfigurowany więc się nie pluje